synapse/storage/event_push_actions.py

```diff
@@ -105,33 +107,38 @@ def _set_push_actions_for_event_and_users_txn(self, txn, event):
             WHERE event_id = ?
         """
 
-        txn.execute(sql, (
-            event.room_id, event.internal_metadata.stream_ordering,
-            event.depth, event.event_id,
-        ))
-
-        user_ids = self._simple_select_onecol_txn(
-            txn,
-            table="event_push_actions_staging",
-            keyvalues={
-                "event_id": event.event_id,
-            },
-            retcol="user_id",
-        )
+        if events_and_contexts:
```

Comment:
why do we do this check?

Reply:
I'm not 100% convinced that we `events_and_contexts` can't be empty, given the number of times we filter it down. The rest of the code should be fine with it being empty.

Reply:
and why is this code not ok with it being empty?

Reply:
Maybe it is? But there isn't any documentation that `executemany` is fine with empty lists and its the sort of thing I can imagine it complaining about.